volatility/framework/plugins/windows/netlist.py

```diff
@@ -0,0 +1,544 @@
+
+                    entry_offset = context.symbol_space.get_type(obj_name).relative_child_offset("ListEntry")
+                    endpoint = context.object(obj_name, layer_name = layer_name, offset = endpoint_entry - entry_offset)
+                    yield endpoint
+
+    @classmethod
+    def create_tcpip_symbol_table(cls,
+                                    context: interfaces.context.ContextInterface,
+                                    config_path: str,
+                                    layer_name: str,
+                                    tcpip_module: interfaces.objects.ObjectInterface) -> str:
+        """Creates
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            config_path: The config path where to find symbol files
+            layer_name: The name of the layer on which to operate
+            tcpip_module: The created vol Windows module object of the given memory image
+
+        Returns:
+            The name of the constructed and loaded symbol table
+        """
+        guids = list(
+            PDBUtility.pdbname_scan(
+                context,
+                layer_name,
+                context.layers[layer_name].page_size,
+                [b"tcpip.pdb"],
+                start=tcpip_module.DllBase,
+                end=tcpip_module.DllBase + tcpip_module.SizeOfImage
+            )
+        )
+
+        if not guids:
+            raise exceptions.VolatilityException("Did not find GUID of tcpip.pdb in tcpip.sys module @ 0x{:x}!".format(tcpip_module.DllBase))
+
+        guid = guids[0]
+
+        vollog.debug("Found {}: {}-{}".format(guid["pdb_name"], guid["GUID"], guid["age"]))
+
+        return PDBUtility.load_windows_symbol_table(context,
+                                                    guid["GUID"],
+                                                    guid["age"],
+                                                    guid["pdb_name"],
+                                                    "volatility.framework.symbols.intermed.IntermediateSymbolTable",
+                                                    config_path="tcpip")
+
+    @classmethod
+    def find_port_pools(cls,
+                       context: interfaces.context.ContextInterface,
+                       layer_name: str,
+                       net_symbol_table: str,
+                       tcpip_symbol_table: str,
+                       tcpip_module_offset: int,
+                       pointer_length: int) -> (int, int):
+        """Finds the given image's port pools. Older Windows versions (presumably < Win10 build 14251) use driver
+        symbols called `UdpPortPool` and `TcpPortPool` which point towards the pools.
+        Newer Windows versions use `UdpCompartmentSet` and `TcpCompartmentSet`, which we first have to translate into
+        the port pool address. See also: http://redplait.blogspot.com/2016/06/tcpip-port-pools-in-fresh-windows-10.html
```

Comment:
Yes! I wish I could include more references, but almost all structures I use in this plugin are not documented anyplace I could find unfortunately.

Reply:
Sure, but just providing any references at all is awesome!  This was one of those complimentary comments I try to include (otherwise reviews can come across quite negative).  5:D